src/lib.rs

```diff
@@ -346,172 +191,66 @@ pub trait PolynomialCommitment<F: Field>: Sized {
     /// presuming that commitments and query_set is already bound to the internal
     /// state of the Fiat-Shamir rng.
     /// TODO: rename this function
-    fn check_combinations_individual_opening_challenges<'a>(
+    fn multi_point_multi_poly_verify<'a>(
         vk: &Self::VerifierKey,
-        linear_combinations: impl IntoIterator<Item = &'a LinearCombination<F>>,
         commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
-        eqn_query_set: &QuerySet<F>,
-        eqn_evaluations: &Evaluations<F>,
-        proof: &BatchLCProof<F, Self>,
+        query_set: &QuerySet<G::ScalarField>,
+        evaluations: &Evaluations<G::ScalarField>,
+        proof: &Self::MultiPointProof,
         fs_rng: &mut Self::RandomOracle,
     ) -> Result<bool, Self::Error>
         where
-            Self::Commitment: 'a,
-    {
-        let BatchLCProof { proof, evals } = proof;
-
-        if !evals.is_some() {
-            Err(Error::IncorrectProof)?
-        }
-
-        let lc_s = BTreeMap::from_iter(linear_combinations.into_iter().map(|lc| (lc.label(), lc)));
-
-        let poly_query_set = lc_query_set_to_poly_query_set(lc_s.values().copied(), eqn_query_set);
-        let poly_evals = Evaluations::from_iter(
-            poly_query_set
-                .iter()
-                .map(|(_, point)| point)
-                .cloned()
-                .zip(evals.clone().unwrap()),
-        );
-
-        for &(ref lc_label, (_, point)) in eqn_query_set {
-            if let Some(lc) = lc_s.get(lc_label) {
-                let claimed_rhs = *eqn_evaluations.get(&(lc_label.clone(), point)).ok_or(
-                    Error::MissingEvaluation {
-                        label: lc_label.to_string(),
-                    },
-                )?;
-
-                let mut actual_rhs = F::zero();
-
-                for (coeff, label) in lc.iter() {
-                    let eval = match label {
-                        LCTerm::One => F::one(),
-                        LCTerm::PolyLabel(l) => *poly_evals
-                            .get(&(l.clone().into(), point))
-                            .ok_or(Error::MissingEvaluation { label: l.clone() })?,
-                    };
-
-                    actual_rhs += &(*coeff * eval);
-                }
-                if claimed_rhs != actual_rhs {
-                    eprintln!("Claimed evaluation of {} is incorrect", lc.label());
-                    return Ok(false);
-                }
-            }
-        }
-
-        let pc_result = Self::batch_check_individual_opening_challenges(
-            vk,
-            commitments,
-            &poly_query_set,
-            &poly_evals,
-            proof,
-            fs_rng,
-        )?;
-
-        if !pc_result {
-            eprintln!("Evaluation proofs failed to verify");
-            return Ok(false);
-        }
-
-        Ok(true)
-    }
+            Self::Commitment: 'a;
```

Comment:
We decided to put the implementation for this function in the trait.
As it is now the trait of a linear polynomial commitment scheme, that is the right place.

Reply:
Done